linker/linker_test.go

```diff
@@ -2954,7 +2954,28 @@ func TestLinkerValidation(t *testing.T) {
 					}
 				`,
 			},
-			expectedErr: `test.proto:14:25: expected extension with number 1 to be named .foo.bar, not foo.s, per declaration at test.proto:8:25`,
+			expectedErr: `test.proto:14:25: expected extension with number 1 to be named foo.bar, not foo.s, per declaration at test.proto:8:25`,
+		},
+		"failure_extension_name_does_not_match_declaration2": {
```

Comment:
One case we're not explicitly testing for is an extension range with `verification=DECLARATION` with an undeclared extension. It errors as expected, not sure if you think it's worth covering, since some logic is now based on whether extension declarations are empty.

Reply:
I think we already have a case for that, unless I am misunderstanding you.

Test case "failure_extension_type_does_not_match_declaration", below on line 2980. (This PR adds one below it that is the same, but _without_ the `verification=DECLARATION`.)

Reply:
I'm going to merge this now. But if I am mistaken, please elaborate about the missing test case and I will happily add it in a follow-up PR.

Reply:
We're not explicitly testing this case:

```
syntax = "proto2";
package foo;
message A {
	extensions 1 [
		verification=DECLARATION
	];
}
extend A {
	optional uint32 bar = 1;
}
```

I'm not sure we're _missing_ it. This depends on how we want to test. I'd consider adding the test to guard against a bad refactor that only activates verification if declarations are present (which would pass all other tests).